cpp/include/cudf/strings/strings_column_handler.hpp

```diff
@@ -0,0 +1,232 @@
+                                      cudf::column_view gather_map,
+                                      cudaStream_t stream=(cudaStream_t)0 );
+
+/**---------------------------------------------------------------------------*
+ * @brief Returns a new strings column that is a sorted version of the
+ * strings in this instance.
+ *
+ * @param stype Specify what attribute of the string to sort on.
+ * @param ascending Sort strings in ascending or descending order.
+ * @param nullfirst Sort nulls to the beginning or the end of the new column.
+ * @param stream CUDA stream to use kernels in this method.
+ * @return New strings column with sorted elements of this instance.
+ *---------------------------------------------------------------------------**/
+std::unique_ptr<cudf::column> sort( strings_column_handler handler,
+                                    strings_column_handler::sort_type stype,
+                                    bool ascending=true,
+                                    bool nullfirst=true,
+                                    cudaStream_t stream=(cudaStream_t)0 );
+
+/**
+ * @brief Returns new instance using the provided map values and strings.
+ * The map values specify the location in the new strings instance.
+ * Missing values pass through from the handler instance into those positions.
+ *
+ * @code
+ * s1 = ["a", "b", "c", "d"]
+ * s2 = ["e", "f"]
+ * map = [1, 3]
+ * s3 = scatter( s1, s2, m1 )
+ * s3 is ["a", "e", "c", "f"]
+ * @endcode
+ *
+ * @param[in] strings The instance for which to retrieve the values
+ *            specified in map column.
+ * @param[in] scatter_map The 0-based index values to retrieve from the
+ *            strings parameter. Number of values must equal the number
+ *            of elements in strings pararameter (strings.size()).
+ * @param stream CUDA stream to use kernels in this method.
+ * @return New instance with the specified strings.
+ */
+std::unique_ptr<cudf::column> scatter( strings_column_handler handler,
+                                       strings_column_handler strings,
+                                       cudf::column_view scatter_map,
+                                       cudaStream_t stream=(cudaStream_t)0 );
+/**
+ * @brief Returns new instance using the provided index values and a
+ * single string. The map values specify where to place the string
+ * in the new strings instance. Missing values pass through from
+ * the handler instance at those positions.
+ *
+ * @code
+ * s1 = ["a", "b", "c", "d"]
+ * map = [1, 3]
+ * s2 = scatter( s1, "e", m1 )
+ * s2 is ["a", "e", "c", "e"]
+ * @endcode
+ * 
+ * @param[in] string The string to place in according to the scatter_map.
+ * @param[in] scatter_map The 0-based index values to place the given string.
+ * @return New instance with the specified strings.
+ */
+std::unique_ptr<cudf::column> scatter( strings_column_handler handler,
+                                       const char* string,
```

Comment:
Is this in host or device memory? Either way, we shouldn't be passing raw pointers for inputs.

For `libcudf++v2.0` I need to work on a `cudf::scalar` object that represents a single element.